ir/type.cpp

```diff
@@ -622,6 +630,30 @@ StateValue AggregateType::extract(const StateValue &val,
                             (val.non_poison << idx).extract(h, l) });
 }
 
+StateValue AggregateType::update(const StateValue &val,
+                                 const StateValue &n,
+                                 const expr &index) const {
+  auto &elementTy = *children[0];
```

Comment:
what happens if Agg has no children?  Can't happen with vectors but can happen with structs.
Either this is moved to struct type or it needs to handle the zero children case

Reply:
extract() has the same "always have children" assumptions. I can move both of them if you agree:)

Reply:
Nicely spot; extract (with non-constant idx) has the same problem.  I think it makes sense to move both to VectorType. None of these methods works with structs, and I don't think we need them for arrays(?)

Reply:
Okay, I'll move both of them to the VectorType class.

Reply:
done